src/main/java/com/paneedah/weaponlib/SyncManager.java

```diff
@@ -27,16 +27,20 @@ public SyncManager(PermitManager permitManager) {
 	}
 	
 	private void syncOnServer(Permit<S> permit, PlayerItemInstance<S> instance) {
-	    LOG.debug("Syncing {} in state {} on server", instance, instance.getState());
+	    //LOG.debug("Syncing {} in state {} on server", instance, instance.getState());
         ItemStack itemStack = instance.getItemStack();
         if(itemStack != null) {
             if(instance.getItem() == itemStack.getItem()) {
-                LOG.debug("Stored instance {} of {} in stack {}", instance, instance.getItem(), itemStack);
+                //LOG.debug("Stored instance {} of {} in stack {}", instance, instance.getItem(), itemStack);
                 instance.reconcile();
-                if(instance.shouldHaveInstanceTags()) Tags.setInstance(itemStack, instance);
+
+				//Is this "CSS"?
+				LOG.debug("\n - {} shouldHaveInstanceTags is {}", instance.getItem(), instance.shouldHaveInstanceTags());
+                if(instance.getItem() instanceof Weapon)
+					Tags.setInstance(itemStack, instance);
```

Comment:
Have you read the code behind `shouldHaveInstanceTags`?

There is nothing `Weapon` specific, lots of things that are not a `Weapon` this change would just break things.

Reply:
It won't damage anything. I've seen shouldHaveInstanceTags, and the code inside will only be executed when the instance is a weapon

Reply:
> It won't damage anything. I've seen shouldHaveInstanceTags, and the code inside will only be executed when the instance is a weapon

That's plain false, it would execute when the instance is a:
- `PlayerGrenadeInstance`
- `PlayerHandheldInstance`
- `PlayerMeleeInstance`
- `PlayerTabletInstance`
- `PlayerWeaponInstance`

Reply:
I forgot these lol